tools/mason/MasonTest.chpl

```diff
@@ -90,28 +90,45 @@ private proc runTests(show: bool, run: bool, parallel: bool) {
     // Check for tests to run
     if numTests > 0 {
 
-      coforall test in testNames {
+      var resultDomain: domain(string);
+      var testResults: [resultDomain] string;
+      var numPassed = 0;
+
+      forall test in testNames with (+ reduce numPassed) {
 
         const testPath = "".join(projectHome, '/test/', test);
-        const testName = basename(test.strip(".chpl"));
+        const testName = basename(stripExt(test, ".chpl"));
 
         // get the string of dependencies for compilation
         const compopts = getDependencyString(sourceList, testName);
 
         const moveTo = "-o " + projectHome + "/target/test/" + testName;
         const compCommand = " ".join("chpl",testPath, projectPath, moveTo, compopts);
         const compilation = runWithStatus(compCommand);
-
         if show then writeln(compCommand);
+
         if compilation != 0 {
           writeln("compilation failed for " + test);
         }
         else {
           if show || !run then writeln("compiled ", test, " successfully");
+          if parallel {
+            var result = runTestBinary(projectHome, testName, show);
+            if result != 0 {
+              testResults[testName] = "Failed";
```

Comment:
FWIW, this is racy, such that if another iteration were to write to the same `testName` element, it would overwrite the previous value. In practice, this should never happen. However, if you wanted to write this safer, you could only modify the domain within this loop, such that the inclusion of a test name in the domain means the test failed, and all the other tests passed.